esrally/mechanic/supplier.py

```diff
@@ -174,6 +184,30 @@ def _src_dir(cfg, mandatory=True):
                                           " all prerequisites and reconfigure Rally with %s configure" % PROGRAM_NAME)
 
 
+class TemplateRenderer:
+    def __init__(self, version, os_name=None, arch=None):
+        self.version = version
+        if os_name is not None:
+            self.os = os_name
+        else:
+            self.os = sysstats.os_name().lower()
+        if arch is not None:
+            self.arch = arch
+        else:
+            self.arch = sysstats.cpu_arch().lower()
+
+    def render(self, template):
+        substitutions = {
+            "{{VERSION}}": self.version,
```

Comment:
Is there a reason -- apart from simplicity -- we can't use j2 template rendering to substitute those (and at the same time be more lenient with spaces like `{{ VERSION }}` and allow filters)? Many other files in rally-teams are treated as j2 hence the question.

Reply:
For other templates (like `elasticsearch.yml` or `jvm.options`) I think Jinja makes sense because their purpose is to serve as templates. `config.ini` is a high-level file meant to control how Rally behaves and I would not expect much templating happening here. As discussed elsewhere, I'll raise a separate enhancement issue.

Reply:
I've raised https://github.com/elastic/rally/issues/856.